### DISCUSSION TITLE
Developer contribution

### DISCUSSION THREAD

**Discussion by axelkunz** [ChristopherKlay/StadiaEnhanced#201] [General]
Hey Chris,

fellow German developer here.

I made some improvements to the code. Mostly refactoring the stream monitor code for improved readability and easier understanding of the logic. This should make it easier to add features/improvements in the future:

- extracted streaming monitor (full/minified) into a separate class 
- extracted menu-attached streaming monitor into a separate class
- added some expressive methods and names, for replacing hard-to-understand if-conditions
- improved double click on monitor to switch state instantly (full <-> minified) instead of waiting for next loop

There is no way for me to create a pull request with my changes. What would be the easiest way to share the code with you, so you can review?

**Comment by ChristopherKlay** [comment]
If you for whatever reason can't create a pull request, feel free to just attach the files in a `.zip` file directly here in this conversation, by just dropping it into the comment field. Checking the differences shouldn't be too complicated (:

**Reply by axelkunz** [reply]
My mistake. I tried to push to a new branch in this repository instead of forking the repository first. Created the pull request: https://github.com/ChristopherKlay/StadiaEnhanced/pull/204
